spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/ModuleRegistrar.java

```diff
@@ -31,19 +37,41 @@
  * @author Marius Bogoevici
  * @author Dave Syer
  */
-public class ModuleRegistrar implements ImportBeanDefinitionRegistrar {
+@Configuration
+public class ModuleRegistrar implements ImportBeanDefinitionRegistrar, EnvironmentAware {
+
+	public static final String SPRING_CLOUD_STREAM_BINDINGS_PREFIX = "spring.cloud.stream.bindings";
+
+	private ConfigurableEnvironment environment;
+
+	@Override
+	public void setEnvironment(Environment environment) {
+		if (environment instanceof ConfigurableEnvironment) {
+			this.environment = (ConfigurableEnvironment) environment;
+		}
+	}
 
 	@Override
 	public void registerBeanDefinitions(AnnotationMetadata metadata,
 			BeanDefinitionRegistry registry) {
 		MultiValueMap<String, Object> attributes = metadata.getAllAnnotationAttributes(
 				EnableModule.class.getName(), false);
+		List<String> registeredChannelNames = new ArrayList<>();
 		for (Class<?> type : collectClasses(attributes.get("value"))) {
-			MessageChannelBeanDefinitionRegistryUtils.registerChannelBeanDefinitions(type, registry);
+			registeredChannelNames.addAll(MessageChannelBeanDefinitionRegistryUtils.registerChannelBeanDefinitions(type, registry));
 			MessageChannelBeanDefinitionRegistryUtils.registerChannelsQualifiedBeanDefinitions(
 					ClassUtils.resolveClassName(metadata.getClassName(), null), type,
 					registry);
 		}
+		Properties defaultChannelNameProperties = new Properties();
+		for (String registeredChannelName : registeredChannelNames) {
+			defaultChannelNameProperties.put(SPRING_CLOUD_STREAM_BINDINGS_PREFIX + "." + registeredChannelName,
+					"${spring.application.name:spring.cloud.stream}" + "." + registeredChannelName);
+		}
+		if (environment != null) {
```

Comment:
Based on [this] (https://github.com/spring-projects/spring-boot/blob/master/spring-boot/src/main/java/org/springframework/boot/SpringApplication.java#L305), I think we can get rid of the null check and directly assign the environment to `ConfigurableEnvironment`.